src/embind/embind.js

```diff
@@ -892,11 +892,14 @@ var LibraryEmbind = {
     }
 
     if (returns) {
-        invokerFnBody += "var ret = retType.fromWireType(rv);\n" +
+        invokerFnBody += "var ret = retType.fromWireType(rv);\n";
 #if EMSCRIPTEN_TRACING
-                         "Module.emscripten_trace_exit_context();\n" +
+        invokerFnBody +=  "Module.emscripten_trace_exit_context();\n";
 #endif
-                         "return ret;\n";
+        if (deleteLater) {
+            invokerFnBody += "ret.deleteLater();"
```

Comment:
Correct, the deleteLater() is so the iterator will be freed somehow.  Since the iterator creation is all handled by JavaScript I could not find another way to make sure it got cleaned up.  I suppose the container could maintain a list of created iterators somehow, and they could be freed at a later time of the users choosing?